Revit_Core_Engine/Convert/Physical/ToRevit/Rebar.cs

```diff
@@ -0,0 +1,136 @@
+        {
+            if (bar == null || document == null)
+                return null;
+
+            Rebar rebar = refObjects.GetValue<Rebar>(document, bar.BHoM_Guid);
+            if (rebar != null)
+                return rebar;
+
+            settings = settings.DefaultIfNull();
+
+            //getting host
+            Element host = null;
+            if (bar.CustomData.ContainsKey("host"))
+                host = document.GetElement(new ElementId((int)bar.CustomData["host"]));
+            else
+            {
+                BH.Engine.Reflection.Compute.RecordError("One or more rebars does not contain information about the host.");
+                return null;
+            }
+
+            if (host == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError($"Rebar host with ID: {bar.CustomData["host"]} does not exist in the Revit project.");
+                return null;
+            }
+
+            //getting bar type
+            RebarBarType barType = bar.ElementType(document, settings);
+            if (barType == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Revit project does not contain rebar family containing type with matching diameter for one or more rebars.");
+                return null;
+            }
+
+            //setting bend radiuses
+            barType.SetParameter(BuiltInParameter.REBAR_STANDARD_BEND_DIAMETER, bar.BendRadius * 2);
+            barType.SetParameter(BuiltInParameter.REBAR_STANDARD_HOOK_BEND_DIAMETER, bar.BendRadius * 2);
+            barType.SetParameter(BuiltInParameter.REBAR_BAR_STIRRUP_BEND_DIAMETER, bar.BendRadius * 2);
+
+            //creating rebar
+            RebarFreeFormValidationResult rffvr = new RebarFreeFormValidationResult();
+
+            List<List<Curve>> curves = new List<List<Curve>> { bar.CentreCurve.IToRevitCurves() };  //
+            IList<IList<Curve>> iListCurves = new List<IList<Curve>>();                             // Needed to convert
+            foreach (List<Curve> list in curves)                                                    // List<List<Curve>> -> IList<IList<Curve>>
+                iListCurves.Add(list);                                                              //
+
+            rebar = Rebar.CreateFreeForm(document, barType, host, iListCurves, out rffvr);
+
+            //setting hooks
+            if (bar.GetType() == typeof(Stirrup))
+            {
+                RebarHookType rht = new FilteredElementCollector(document).OfClass(typeof(RebarHookType)).Where(x => x.Name == "Standard - 90 deg.").FirstOrDefault() as RebarHookType;
+                rebar.SetHookTypeId(0, rht.Id);
+                rebar.SetHookTypeId(1, rht.Id);
+                rebar.SetHookOrientation(0, RebarHookOrientation.Left);
+                rebar.SetHookOrientation(1, RebarHookOrientation.Left);
```

Comment:
Assumption that `Left` will always be inwards seems to be dangerous. However, I tried to break it myself, without success 🐒 So happy to leave as is, until it starts breaking.

Reply:
To be resolved in #987